### ISSUE TITLE
Large values for train loss

### ISSUE THREAD

**Issue by katayooneshkofti** [mariusmerkle/TL-PINNs#1]
Hi @mariusmerkle 
since I saw your contributions to Deepxde, I will be grateful if you have any suggestions for my issue. The problem has two inputs (r as radial distance and t as time) and 2 outputs as displacement and temperature. the boundary and initial conditions are as follow: 
![1](https://user-images.githubusercontent.com/68455679/114660688-fb858f00-9d0a-11eb-8532-c76eacf66c24.JPG)
I defined these boundary and initial conditions as follow: 
`geom = dde.geometry.Interval(1.0, 1.5)
    timedomain = dde.geometry.TimeDomain(0,0.5)
    geomtime = dde.geometry.GeometryXTime(geom, timedomain)
        

    # boundary definition
    
    def boundary_rin (x, on_boundary):
        return on_boundary and np.isclose(x[0], 1.)  #rmin=1
    
    def boundary_rout (x, on_boundary):
        return on_boundary and np.isclose(x[0], 1.5)  #rmax=1.5  
    
    def boundary_for_initial(x, _ ):
         return np.isclose(x[-1],0)
          
    def func(x):
        return np.zeros((len(x),1), dtype=np.float32)
   
    def normal_dist(x , mean , sd):
        prob_density = (1/(np.sqrt(2*np.pi)*sd)) * np.exp(-0.5*((x-mean)/sd)**2)
        return prob_density 
    
    mean=0.5
    sd=0.04
        
    def Tstar(x):
        return normal_dist(x[:,1:2], mean, sd)
    
    def dU_t(x,y):
        U, T = y[:,0:1], y[:,1:2]  #Defining Outputs
        r, t = x[:,0:1], x[:,1:2]
        dU_t = dde.grad.jacobian (y , x , i=0 , j=1)
        return dU_t
    def dT_t(x,y):
        U, T = y[:,0:1], y[:,1:2]  #Defining Outputs
        r, t = x[:,0:1], x[:,1:2]
        dT_t = dde.grad.jacobian (y , x , i=1 , j=1)
        return dT_t  
    
       
    bc1= dde.DirichletBC(geomtime,func,boundary_rin, component=0)
    bc2= dde.DirichletBC(geomtime,func,boundary_rout,component=0)
    bc3=dde.DirichletBC(geomtime,Tstar, boundary_rin , component=1)
    bc4=dde.DirichletBC(geomtime,func,boundary_rout,component=1)
    ####Considering Neumann Boundary Conditions for initial conditions with derivatives###
    bc5=dde.OperatorBC(geomtime, lambda x,y,_: dU_t(x,y), boundary_for_initial) 
    bc6=dde.OperatorBC(geomtime, lambda x,y,_: dT_t(x,y), boundary_for_initial) 

    #Initial Conditions
    ic1= dde.IC(geomtime, func, boundary_for_initial, component=0)
    ic2= dde.IC(geomtime, func, boundary_for_initial, component=1)   
    
   
    
    data = dde.data.TimePDE(geometryxtime=geomtime, pde=pde_system, ic_bcs=[bc1, bc2, bc3, bc4, bc5, bc6, ic1, ic2],  num_domain=10000, num_boundary=5000, num_initial=5000, num_test=100)
    layer_size = [2] + [20] * 4 + [2] 
    activation = "tanh"
    initializer = "Glorot uniform"   ###Glorot normal
    net = dde.maps.FNN(layer_size, 'LAAF10 tanh' , initializer) ###Adaptive Activation Function: dde.maps.activations.layer_wise_locally_adaptive
    #Transform the output 
    net.apply_output_transform(
        lambda x, y : tf.concat(
            [
            (1-x[: , 0:1]) * (1.5-x[: , 0:1]) *x[: , 1:2] * y[: , 0:1],
             (1.5-x[: , 0:1]) * y[ : , 1:2] * x[: , 1:2]
            ], axis =1
            )   )   
                                                                                                        
    model = dde.Model(data, net)
    #dde.postprocessing.plot_boundary_conditions(data)
    model.compile("adam", lr=1e-3 )
    #checkpointer = dde.callbacks.ModelCheckpoint("./model/model.ckpt", verbose=1, save_better_only=True)
    losshistory, train_state = model.train(epochs=10000)#, model_restore_path = "./model/model.ckpt-5000", callbacks = [checkpointer])
    model.compile("L-BFGS-B")
    dde.saveplot(losshistory, train_state, issave=True, isplot=True)`

I used Hard BCs, Weighting the Loss terms, and assigning more points to num_domain and boundary/initial_points, but none of them helped. the loss term is not acceptable especially for the boundary condition with T*

Do you have any suggestions? thank you for your consideration..

**Comment by mariusmerkle** [comment]
Hi @katayooneshkofti,

I'm happy to help you. Could you please **clarify the following three issues**:

1. The code doesn't contain the pde_system -- what's the **system of differential equations** you're trying to solve?
2. The **boundary and initial conditions in T contradict each other**: At x = r_in and t = t_0, the initial condition enforces T(r = r_in, t = t_0) = 0 while the boundary condition enforces T(r = r_in, t = t_0) = T_0^*/(sigma^*sqrt(2*pi)) which is not equal to zero. This is not solvable.
3. What do you mean with '**the loss term is not acceptable**'? Is the value of the loss function too high, i.e. not converged?
